app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java

```diff
@@ -78,6 +78,8 @@ public void onBackPressed() {
             binding.drawerLayout.closeDrawer(GravityCompat.START);
         } else if (fragmentNavigator.isDashboardActive()) {
             backPressHandler.onBackPressed(this, super::onBackPressed);
+        } else if (getSupportFragmentManager().getBackStackEntryCount() == 2) {
+            getSupportFragmentManager().popBackStack();
```

Comment:
This is magic case. We don't want this. Why is this special case needed at all? All of this should be handled in fragmentNavigator

Reply:
This special case was needed because when we I am adding the CreateEventFragment to backstack and on pressing the back button from this fragment, the dashboard would come up instead of the EventsListFragment because of this code in the onBackPressed( ) 

`  binding.navView.getMenu().findItem(R.id.nav_dashboard).setChecked(true);
            getSupportActionBar().setTitle(R.string.dashboard);
`

Reply:
But this is very risky case and hence will cause bugs in future, the fragmentNavigator should already pop backstack until it is empty. The entryCount == 2 has no logical reasoning to go with the event create fragment. Either this needs to be handled by fragmentNavigator transparently, or another solution is needed

Reply:
What if I add the following to the back( ) of FragmentNaigator

       if (fragmentManager.getBackStackEntryCount() > 1) {
           fragmentManager.popBackStack(); 
           dashboardActive = false;
        } else {
            fragmentManager.popBackStack();
            lastSelectedNavItemId = R.id.nav_dashboard;
            dashboardActive = true;
        }
    }

Reply:
```Java
if (fragmentManager.getBackStackEntryCount() > 1) {
       fragmentManager.popBackStack();
```
This should already include 

`if (fragmentManager.getBackStackEntryCount() == 2)`

Reply:
And that logic will unconditionally pop backstack

Reply:
Ok.I'll look for another solution then.

Reply:
Can I create an Activity for Events List instead of the current EventListFragment and then use the CreateEventFragment in this Activity? Then this fragment will become separate from the Dashboard

Reply:
No, this will become very complex. Also, we are pushing towards having only one activity

Reply:
I have added a TAG like the following when the CreateNewFragment is added

public void openCreateEventFragment() {

        getFragmentManager().beginTransaction()
                            .add(R.id.fragment_container, CreateEventFragment.newInstance(), "CreateEventFragment")
                            .addToBackStack(null)
                            .commit();
    }

and in the onBackPressed( ) of MainActivity added the following condition:

    public void onBackPressed() {

        if (binding.drawerLayout.isDrawerOpen(GravityCompat.START)) {
            binding.drawerLayout.closeDrawer(GravityCompat.START);
        } else if (fragmentNavigator.isDashboardActive()) {
            backPressHandler.onBackPressed(this, super::onBackPressed);
        } **else if (getSupportFragmentManager().findFragmentByTag("CreateEventFragment") != null) {
            getSupportFragmentManager().popBackStack();
        }** else {
            fragmentNavigator.back();
            binding.navView.getMenu().findItem(R.id.nav_dashboard).setChecked(true);
            getSupportActionBar().setTitle(R.string.dashboard);
        }
    }

I guess this will be a safer operation.

Reply:
This is unneeded dynamic check. This is not a good solution because it is not general. This is relying too much on checks like this. This will surely break if behaviour of navigator is changed. You'll have to add 5 more such checks when new fragments for session, speaker and other details are added, So saying this, it is a sloppy workaround, not a fix. I know you might be thinking if it is working why not just use it, because this will solve one issue and create 100 more in future.

The error you are facing may be solved by adding 2-3 lines of 'correct' logical code in fragment navigator. This is because one of the things is broken and we are patching wrong stuff. I don't know what is broken, or would have told you so, but this is definitely not the right solution

Any kind of dynamic check on CreateEventFragment is a no go. This should be a general fix